Lib/test/test_capi/test_unicode.py

```diff
@@ -1776,6 +1776,12 @@ def test_utf8(self):
         self.assertEqual(writer.finish(),
                          "ascii-latin1=\xE9-euro=\u20AC.")
 
+    def test_ascii(self):
+        writer = self.create_writer(0)
+        writer.write_ascii(b"Hello ", -1)
```

Comment:
Not sure if this is the place, but do you want to test empty strings as well? (we might already cover this test case elsewhere)

Reply:
I added a test on an empty string.